src/lib.rs

```diff
@@ -172,6 +179,9 @@ mod test {
             author: Address("no-reply@domain.com".to_owned()),
         });
 
+        let mut balances: BTreeMap<String, u16> = BTreeMap::new();
```

Comment:
Okay, so this test will show `parse(serialize(x)) == x` right?

And does work with `u16` as tree value, which is promising. 

Maybe other pieces already work. Can you add test cases for other types? 

Also, this only works when the map has all the same type of value, right? I guess we need some more work for flatten, but we can leave that for another PR.